pkg/eks/podidentities/plan.go

```diff
@@ -0,0 +1,102 @@
+/*
```

Comment:
Thanks for the feedback. A lot of this was derived from following another change by richcase. I am totally unopposed to the changes suggested, but do we have a styleguide or the likes which dictate where/when/how we should be making these changes? Using/not using plans, structured logging/eventing etc.

Reply:
We don't have a style or coding guide, I think. I did a regular code review and while reading, I found this to be a bit too complex for what it's doing, especially since one needs to read multiple functions. The source example `pkg/eks/addons/plan.go` is doing a bit more because of addon "states", and didn't have access to the nice `slices` module at the time. I don't think the code is _very_ bad here. But maybe we can modernize rather than copy the existing idea as-is. It can be mostly a single function.